File: TostiTime.FrontEnd/src/app/_models/office.ts
import { Iron } from "./iron";
import { Person } from "./person";

export interface Office {
    id: number;
    country: string;
    city: string;
    cityCode: string;
    address: string;
    postalCode: string;
    numberOfIrons: number;
    irons: Iron[];
    persons: Person[];
}

File: TostiTime.Core/Entities/Office.cs
namespace TostiTime.Core.Entities;

public class Office : EntityBase
{
    public string Country { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string CityCode { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public List<Person> Persons { get; set; } = new();
    public int NumberOfPersons => Persons.Count;
    public List<Iron> Irons { get; set; } = new();
    public int NumberOfIrons => Irons.Count;
}
